fw/http.c

```diff
@@ -5857,7 +5857,7 @@ tfw_http_req_process(TfwConn *conn, TfwStream *stream, struct sk_buff *skb)
 		T_DBG2("Block invalid HTTP request\n");
 		TFW_INC_STAT_BH(clnt.msgs_parserr);
 		tfw_http_req_parse_drop(req, 400, "failed to parse request");
-		return TFW_BLOCK;
```

Comment:
@krizhanovsky  @EvgeniiMekhanik  Parsing errors should not be considered as security event?

Reply:
We filter a lot of attack on the parser level (e.g. injection attacks). However, at the moment we don't distinguish innocent parser errors from  clear attacks and just return `TFW_BLOCK` everywhere. I'd prefer to reply by default with something telling to a user, than just brutally reset connection in many cases. So let's treat parser errors as just errors, not attacks for now.

It seems we need to fix the header comment for `tfw_http_req_parse_drop()` since it states "Caller must return TFW_BLOCK...", which isn't consistent with the current code.

Reply:
Fixed

Reply:
I believe, we should also check for `T_BAD` now inside `tfw_h2_frame_process()`. 
For example here:
```
(fw/http_frame.c:1880) 
		h2->data_off = 0;
		h2->skb_head = data_up.skb->next = data_up.skb->prev = NULL;
		r = tfw_http_msg_process_generic(c, h2->cur_stream, data_up.skb);
		if (r == T_DROP) {
			kfree_skb(nskb);
			goto out;
		}
	} else {
```

Reply:
Yes, and we also should check for all `r` values not only T_BAD, because `tfw_http_msg_process_generic` can return also system errors.